app/models/user_counter.rb

```diff
@@ -4,6 +4,7 @@ class UserCounter < ApplicationRecord
   DECAY_RATE = -0.0077
 
   belongs_to :user
+  validates :counter_name, format: {with: /\A[a-zA-Z0-9_\-.]+\z/, message: "has invalid characters"}
```

Comment:
I wasn't sure about `\-` in a regex character class - I thought you were supposed to mention the hyphen first so it's not interpreted as a range character - but the docs I've read seem to indicate escaping it will work, too.

Reply:
I can move it to be the first character if that's a more standard practice in ruby/rails